stl/inc/condition_variable

```diff
@@ -96,19 +82,10 @@ public:
     template <class _Lock, class _Clock, class _Duration>
     cv_status wait_until(_Lock& _Lck, const chrono::time_point<_Clock, _Duration>& _Abs_time) {
         // wait until time point
-        for (;;) {
-            const auto _Now = _Clock::now();
-            if (_Abs_time <= _Now) {
-                return cv_status::timeout;
-            }
-
-            _CSTD xtime _Tgt;
-            (void) _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
-            const cv_status _Result = wait_until(_Lck, &_Tgt);
-            if (_Result == cv_status::no_timeout) {
-                return cv_status::no_timeout;
-            }
-        }
+        const auto _Now = _Clock::now();
+        _CSTD xtime _Tgt;
+        (void) _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
+        return wait_until(_Lck, &_Tgt);

```

Comment:
This seems to damage correctness by returning `cv_status::timeout` after 10 days if `_Abs_time` is more than 10 days in the future.

Reply:
I think we want to:
```c++
const bool _Clamped = _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
const cv_status _Result = wait_until(_Lck, &_Tgt);
if (_Clamped) {
  return cv_status::no_timeout;
}
return _Result;
```

Reply:
I think we've also damaged correctness by allowing negative durations into `_To_xtime_10_day_clamped` which previously couldn't happen. That function should probably pretend the input is `0` when negative - IIRC all callers treat negatives and 0 the same.

Reply:
Great catch! I wonder if we should make the 10-day clamp configurable for the purposes of manual testing (a 10-second clamp is testable in a way that a 10-day clamp is not).

Reply:
> I wonder if we should make the 10-day clamp configurable for the purposes of manual testing (a 10-second clamp is testable in a way that a 10-day clamp is not).

That would be great. It's nightmarish to test anything timeout-related. @bhardwajs, let us know if that sounds like more work than you signed on for and we'll make a separate issue.

Reply:
@CaseyCarter - I can take on the configurability of `_clamped`. I think it should be a small change. Right?

For tracking purposes, that should be a separate issue of enhancement in infrastructure. Not sure, if we want to have a separate PR for that. What do you think?

Reply:
A separate PR seems best; that way if we ever need to port, modify, or revert the bugfix, we don't have to worry about the test machinery enhancement getting tangled up with it.